src/components/ProfilePage.tsx

```diff
@@ -0,0 +1,221 @@
+            setSummary(bundle.userProfile?.summary)
+            setUrl(bundle.userProfile?.url)
+            setProfilePicUrl(picUrl)
+            setVerfied(bundle.isVerified)
+        } catch(err:any) {
+            displayToast(err.reason as string, 'danger')
+        }
+    }
+
+    const updateUserProfile = async () => {
+        try{
+            if(userProfile){
+                userProfile.firstName = firstName as string
+                userProfile.lastName = lastName as string
+                userProfile.position = position as string
+                userProfile.company = company as string
+                userProfile.location = location as string
+                userProfile.summary = summary as string
+                userProfile.url = url as string
+                userProfile.profilePicureFileHandleId = fileHandleId as string
+                await updateMyUserProfile(userProfile, accessToken)
+                displayToast('Profile has been successfully updated', 'success')
+                getProfile()
+                setEditing(false)
+            }
+        } catch(err:any){
+            displayToast(err.reason as string, 'danger')
+        }
+    }
+
+    useEffect(()=>{
+        getProfile()
+    },[accessToken,])
+
+    const EditField = (label: string, updatedValue: string|undefined, updateFn: Function) => {
+        return(
+            <div className='edit-field'>
+                <FormLabel>{label}</FormLabel>
+                <FormControl
+                    onChange={e=>updateFn(e.target.value)}
+                    value={updatedValue}
+                    placeholder= {`Enter ${label}`}
+                /> 
+            </div>
+        )
+    }
+
+    const onCancel = async () => {
+        getProfile()
+        setEditing(false)
+    }
+
+    const uploadHandler = async(e: React.ChangeEvent<HTMLInputElement>) => {
+        if(e.target.files){
+            const file = e.target.files[0]
+            try{
+                const resp: FileUploadComplete = await uploadFile(accessToken, file.name, file)
+                const picUrl = await getFileHandleByIdURL(resp.fileHandleId,accessToken)
+                setFileHandleId(resp.fileHandleId)
+                setProfilePicUrl(picUrl)
+            } catch(err:any){
+                console.error(err)
+            }
+        }
+    }
+
+    const hiddenFileInput = React.useRef<HTMLInputElement>(null)
+    const clickHandler = () => {
+        if (hiddenFileInput?.current!) {
+          hiddenFileInput.current?.click()
+        }
+    }
+
+    const uploadImg = (
+        <>
+            <input
+                type={'file'}
+                ref={hiddenFileInput}
+                onChange={uploadHandler}
+                style={{display: 'none'}}
+            />
+            <Button style={{marginTop:'16px'}} variant='secondary' onClick={clickHandler}>Upload File</Button>
+        </>
+    )
+
+    const validProfilePic = (
```

Comment:
Not sure about the name "validProfilePic".  Maybe something like "decoratedProfilePic"?